modules/core/src/main/java/org/apache/ignite/internal/processors/failure/FailureProcessor.java

```diff
@@ -137,8 +150,12 @@ public synchronized boolean process(FailureContext failureCtx, FailureHandler hn
                 " WAL path: " + ctx.config().getDataStorageConfiguration().getWalPath() +
                 " WAL archive path: " + ctx.config().getDataStorageConfiguration().getWalArchivePath());
 
-        if (IGNITE_DUMP_THREADS_ON_FAILURE)
-            U.dumpThreads(log);
+        if (IGNITE_DUMP_THREADS_ON_FAILURE) {
+            if (isFailureIgnored(failureCtx, hnd))
+                U.dumpThreads(log, false);
```

Comment:
It could be rewritten in one line: `U.dumpThreads(log, !isFailureIgnored(failureCtx, hnd));`